tests/test_http_response.py

```diff
@@ -2,14 +2,16 @@
 import unittest
 from unittest import mock
 
+from pkg_resources import parse_version
```

Comment:
Thanks for explanation about URL living standard! I applied the changes suggested, the only thing i had to change is the condition to `xfail`, thanks!

Reply:
My thinking here was to keep the condition I suggested, and change the test expectations to remove %20. So these tests now, instead of testing that spaces are kept and percent-encoded, would test that spaces are stripped.

Reply:
Got it, fixed.